cedar-policy-core/src/parser/err.rs

```diff
@@ -442,6 +449,29 @@ pub mod parse_errors {
 
     use super::*;
 
+    /// Details about a `ExpectedStaticPolicy` error.
+    #[derive(Debug, Clone, Diagnostic, Error, PartialEq, Eq)]
+    #[error("expected a static policy, got a template containing the slot {}", slot.id)]
+    #[diagnostic(help("try removing the template slot(s) from this policy"))]
+    pub struct ExpectedStaticPolicy {
+        /// Slot that was found (which is not valid in a static policy)
+        pub(crate) slot: ast::Slot,
+    }
+
+    impl From<ast::UnexpectedSlotError> for ExpectedStaticPolicy {
+        fn from(err: ast::UnexpectedSlotError) -> Self {
+            match err {
+                ast::UnexpectedSlotError::FoundSlot(slot) => Self { slot },
+            }
+        }
+    }
+
+    /// Details about a `ExpectedTemplate` error.
+    #[derive(Debug, Clone, Diagnostic, Error, PartialEq, Eq)]
+    #[error("expected a template, got a static policy")]
+    #[diagnostic(help("a template should include slot(s) `?principal` or `?resource`"))]
+    pub struct ExpectedTemplate {}
```

Comment:
can we make unit structs like this `non_exhaustive`?  Without `non_exhaustive`, since we have no private fields, users can still construct this struct with `ExpectedTemplate {}`, so adding fields is still breaking.

Reply:
Thanks for pointing that out - I didn't realize it was a problem. I've updated the type to match the approach taken [here](https://github.com/cedar-policy/cedar/blob/8b74d95cba4abe314783a152b34a6b4e9ca39265/cedar-policy/src/api/err.rs#L582) with a `_dummy` field.